### PR TITLE
Web3id revision

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Revise CIS4 client and Web3ID related functionality to new notion of credentials, registry contracts without storage.

Depends on
- [x] https://github.com/Concordium/concordium-base/pull/411

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by eb-concordium — APPROVED** [review]

**Review by annenkov — CHANGES_REQUESTED** [review]

File: src/cis4.rs [hunk 1]
```diff
@@ -261,7 +274,7 @@ impl Cis4Contract {
         parameter_vec.extend_from_slice(&to_sign[REVOKE_DOMAIN_STRING.len()..]);
         let parameter = OwnedParameter::try_from(parameter_vec)?;
 
-        self.update_raw(signer, metadata, "revokeCredentialHolder", parameter)
+        self.update_raw(signer, metadata, "revokeCredentialRevoker", parameter)
```
[the remark this answers is not in the document]
Comment (abizjak) [reply src/cis4.rs:277]: I have renamed the function as `revoke_credential_other`.

File: src/cis4.rs [hunk 2]
```diff
@@ -261,7 +274,7 @@ impl Cis4Contract {
...
             .await
     }
 }
```
[the remark this answers is not in the document]
Comment (abizjak) [reply src/cis4.rs:280]: Should be updated now.

File: src/web3id.rs [hunk 3]
```diff
@@ -135,16 +128,9 @@ pub async fn verify_credential_metadata(
...
-                &entry.credential_info.commitment,
-            ))
-            .map_err(|_| CredentialLookupError::CommitmentParseError {
-                contract,
-                cred_id: owner,
-            })?;
+            let issuer_pk = contract_client.issuer(bi).await?;
 
-            let commitments = CredentialsInputs::Web3 { commitment };
+            let commitments = CredentialsInputs::Web3 { issuer_pk };
```
[the remark this answers is not in the document]
Comment (abizjak) [reply src/web3id.rs:133]: Renamed to `inputs`.

File: src/cis4.rs [hunk 4]
```diff
@@ -218,12 +231,13 @@ impl Cis4Contract {
             .await
     }
 
-    /// Revoke a credential as a revoker.
+    /// Revoke a credential as another party, distinct from issuer or holder.
     ///
-    /// The extra nonce that must be provided is the owner's nonce inside the
-    /// contract. The signature on this revocation message is set to expire at
+    /// The extra nonce that must be provided is the nonce associatedw with the
```
Comment (annenkov) [inline src/cis4.rs:236]:
```suggestion
    /// The extra nonce that must be provided is the nonce associated with the
```

**Pull request merged**